### PR TITLE
Add setting to preview selected tab entry

### PR DESCRIPTION
**Pull request by themilkman**
Hi @robinmalburn,

first of all thanks for your plugin, really like to use it.
In this PR I added the possibility to preview the selected tab entry.

Would be nice to see this patched, if it is looks good to you.

Cheers
Erik

### CODE REVIEW

**Comment by robinmalburn** [comment]
Hi @themilkman,

Thanks so much for the contribution, I love the addition.  Also, huge thanks for adding documentation to go along with the new feature, too.

There's some really tiny little bits of feedback on the PR before I merge it in, but after that, I'd be more than happy to merge this in, and then there's some tiny documentation updates to make myself (I can actually update how packages work on Windows now, which will be a long overdue update to add) and I'll get it updated on Package Control :)

Thanks,
Robin

File: tabfilter.py
Comment (robinmalburn) [inline tabfilter.py:30]: For consistency, do you mind changing this to a hash style comment, i.e. `# save index for later usage` - I try to stick to pep8 and try to stick multi-line strings as documentation blocks for method signatures (or for legit multi-line strings).

File: tabfilter.py
Comment (robinmalburn) [inline tabfilter.py:89]: Now that we have two if conditions based on the `index`, can we turn this into an `if` and an `elif`, to avoid it having to evaluate the condition twice even if it's already passed the first check.  So end result will be something like:
```
if index == -1:
  ...
elif index > -1:
  ...
```

File: tabfilter.py
Comment (robinmalburn) [inline tabfilter.py:43]: I'm going to admit, this is definitely me being incredibly pedantic, but I go out of my way to try and avoid else blocks in code if possible - generally I'm of the mind that it's the default outcome of the `if` not passing, so we shouldn't need to force it into an enclosed block unless there's something amiss.  It's not a hill I'd die on, but it is something I try to code around, 'cause in bigger projects or bigger pieces of code, if/else hell can often be an indicator that something else in the design is shaky.

In this scenario, we can avoid the else by defaulting and overriding the optional parameters we're passing through, so it becomes something more like:

```python
on_highlight_cb = None
selected_index = None

if preview_tab is True and self.window.num_groups() == 1:
    # This doesn't work with a multi group layout
    # but otherwise, replace our defaulted arguments.
    on_highlight_cb = self._on_highlighted
    selected_index = self.current_tab_idx

self.window.show_quick_panel(tabs, self._on_done, on_highlight=on_highlight_cb, selected_index=selected_index)
```

File: tabfilter.py
Comment (robinmalburn) [inline tabfilter.py:31]: Could we define a default for this alongside the other class properties such as `self.prefix` a little higher up in the file?  I'm guessing based on the code later in the event handler, we'd want to default to something like `-1`.  Otherwise, trying to run the plugin with no open tabs results in an `AttributeError`

**Review by robinmalburn — CHANGES_REQUESTED** [review]
As I said in my other comment, the feature is great - it's tested locally and really happy with it.  So the feedback is largely stylistic to keep the consistency with the rest of the code base, with the exception of one edge case bug.

**Comment by themilkman** [comment]
Thank you very much for your kind response and rich feedback! 
I'll happily take a deeper look into it during the next days and see how I may resolve those issues :)

**Comment by themilkman** [comment]
Hi @robinmalburn,

again thank you for your helpful feedback and explanations!
I just adopted them in a new commit. In case that it now fine for you, I could - if you prefer it - squash them (of course you may feel free to do so too).

I am happily looking forward to see a new release of your plugin with this functionality.

Cheers
Erik

**Comment by robinmalburn** [comment]
Hi @themilkman, 

Thanks for making those updates, I really appreciate that.  I'll have another look over them and give it another try locally, but after that, I think it should be good to merge in (I'm happy without the commits squashed, I always like having a full history of commits anyhow).  I've got a couple of other minors tweaks to add myself and then it should be good to get updated on Package Control.

Thanks,
Robin

File: tabfilter.py
Comment (robinmalburn) [inline tabfilter.py:45]: Ahh, sorry, this was a typo on my part in the last comment - `selected_index` has to be an integer, so we need to default this to `-1`, not `None`, otherwise the quick info window won't show at all when working with multiple panes.

**Review by robinmalburn — CHANGES_REQUESTED** [review]
Just one last very last change, which is a bug introduced by the default `selected_index` - turns out, that needs to be an integer, so the default should be `-1` (tested locally to be sure).  The documentation for that is _really_ ambiguous unfortunately.  Though at least the documentation's gotten better over time, it was really lacking back when this was first written.

That should be the last thing, and the last little bits of house keeping I can pick up whilst prepping the release.

**Comment by themilkman** [comment]
Hi @robinmalburn,

sorry for not testing it with the new option disabled... Especially because I should've seen it as I intially tried to initialize current_tab_idx with None and thus also had this error.

I now used 0 as default value because that's what it is shown in the default behaviour, right (first element focussed)? For me -1 would be too much a "magic" preset. What do you think about this? If I am wrong or you've another opinion on it I'll change it to -1 asap. 

Generally I am not still not sure if making highlighting the current tab as in the new preview feature here (where it was necessary) wouldn't be better usability in general. But this is not related to the current pull request, so it doesn't matter for now :)

Thanks again for your time and effort you put in giving so great and constructive feedback!

**Comment by robinmalburn** [comment]
Hi @themilkman,

I've only had chance to have a real quick look tonight, so will give it a proper test tomorrow, but I really like the change to use `0` instead of `-1`.  From a quick test and a re-read of the API spec, that works perfectly and makes way more sense as a default, so that's a really good change.

I'm going to give it one more round of testing when it's not last thing at night, but I think we're in a good place now and I should be good to go ahead and start getting the next release ready including this feature.

Thanks so much for the contribution and for being so good with taking onboard all the feedback.  Really appreciate the work you've put into this feature :)

**Pull request merged**
